dom.go

```diff
@@ -1946,120 +1992,196 @@ func (e *HTMLCanvasElement) GetContext(param string) *js.Object {
+
+func (ctx *CanvasRenderingContext2D) FillText(text string, x, y, maxWidth float64) {
+	if maxWidth == -1 {
+		ctx.Call("fillText", text, x, y)
+		return
+	}
+
+	ctx.Call("fillText", text, x, y, maxWidth)
 }
 
-func (ctx *CanvasRenderingContext2D) StrokeRect(x, y, width, height int) {
-	ctx.Call("strokeRect", x, y, width, height)
+func (ctx *CanvasRenderingContext2D) StrokeText(text string, x, y, maxWidth float64) {
+	if maxWidth == -1 {
+		ctx.Call("strokeText", text, x, y)
+		return
+	}
+
+	ctx.Call("strokeText", text, x, y, maxWidth)
+}
+func (ctx *CanvasRenderingContext2D) MeasureText(text string) *TextMetrics {
+	textMetrics := ctx.Call("measureText", text)
+	return &TextMetrics{Object: textMetrics}
 }
 
-func (ctx *CanvasRenderingContext2D) ClearRect(x, y, width, height int) {
-	ctx.Call("clearRect", x, y, width, height)
+// Line styles
+
+func (ctx *CanvasRenderingContext2D) GetLineDash() interface{} {
+	return ctx.Call("getLineDash")
 }
 
-// Paths
+//
+func (ctx *CanvasRenderingContext2D) SetLineDash(arr interface{}) {
+	ctx.Call("setLineDash", arr)
+	return
+}
 
-func (ctx *CanvasRenderingContext2D) Fill() {
-	ctx.Call("fill")
+// Gradients and patterns
+
+func (ctx *CanvasRenderingContext2D) CreateLinearGradient(x0, y0, x1, y1 float64) {
+	ctx.Call("createLinearGradient", x0, y0, x1, y1)
 }
 
-func (ctx *CanvasRenderingContext2D) Stroke() {
-	ctx.Call("stroke")
+func (ctx *CanvasRenderingContext2D) CreateRadialGradient(x0, y0, r0, x1, y1, r1 float64) {
+	ctx.Call("createRadialGradient", x0, y0, r0, x1, y1, r1)
 }
 
-func (ctx *CanvasRenderingContext2D) BeginPath() {
-	ctx.Call("beginPath")
+func (ctx *CanvasRenderingContext2D) CreatePattern(image *Element, repetition string) {
+	ctx.Call("createPattern", image, repetition)
 }
 
-func (ctx *CanvasRenderingContext2D) MoveTo(x, y int) {
-	ctx.Call("moveTo", x, y)
+// Paths
+
+func (ctx *CanvasRenderingContext2D) BeginPath() {
+	ctx.Call("beginPath")
 }
 
 func (ctx *CanvasRenderingContext2D) ClosePath() {
 	ctx.Call("closePath")
 }
 
+func (ctx *CanvasRenderingContext2D) MoveTo(x, y int) {
+	ctx.Call("moveTo", x, y)
+}
+
 func (ctx *CanvasRenderingContext2D) LineTo(x, y int) {
 	ctx.Call("lineTo", x, y)
 }
 
-func (ctx *CanvasRenderingContext2D) Clip() {
-	ctx.Call("clip")
+func (ctx *CanvasRenderingContext2D) BezierCurveTo(cp1x, cp1y, cp2x, cp2y, x, y int) {
+	ctx.Call("bezierCurveTo", cp1x, cp1y, cp2x, cp2y, x, y)
 }
 
-func (ctx *CanvasRenderingContext2D) QuadraticCurveTo(cpx, cpy, x, y int) {
+func (ctx *CanvasRenderingContext2D) QuadraticCurveTo(cpx, cpy, x, y float64) {
 	ctx.Call("quadraticCurveTo", cpx, cpy, x, y)
 }
 
-func (ctx *CanvasRenderingContext2D) BezierCurveTo(cp1x, cp1y, cp2x, cp2y, x, y int) {
-	ctx.Call("bezierCurveTo", cp1x, cp1y, cp2x, cp2y, x, y)
-}
-
-func (ctx *CanvasRenderingContext2D) Arc(x, y, r int, sAngle, eAngle float64, counterclockwise bool) {
+func (ctx *CanvasRenderingContext2D) Arc(x, y, r, sAngle, eAngle float64, counterclockwise bool) {
```

Comment:
Coordinates can be float64 the mantissa part does actually influence when using drawinf functions.

The only case where i could see the float64 being a problem is in functions that work with array indexes.

That last statement might also be actually false i just tested with DrawImage PutImageData, and GetImageData and it works without issue with floats.